.github/workflows/scripts/scanner-get-released-tags.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+#
+# Print a JSON containing all scanner vulnerability bundle streams, as directed
+# by the `VULNERABILITY_BUNDLE_VERSION` file.
+
+set -euo pipefail
+
+# Save all tags from the repo to a temporary file.
+tags=$(mktemp)
+git tag > "$tags"
+
+# Function to resolve the tag associated with a vulnerability updater version.
+# This logic is duplicated from scanner-output-release-versions.sh. An optimization is needed to eliminate this duplication.
+# TODO: ROX-25999
+resolve_tag() {
+    local ver=${1:?"missing required argument: version"}
+    local tag=${2:?"missing required argument: tag"}
+
+    # If the tag is "master", return it as is
+    if [ "$tag" == "master" ]; then
+        echo "master"
+        return
+    fi
+
+    # Check if the tag exists in the repository
+    if grep -qx "$tag" "$tags"; then
+        echo "$tag"
+        return
+    fi
+
+    echo >&2 "WARNING: Tag '$tag' is not a valid tag in the repository"
+
+    # If the tag is not in the form X.Y.Z, return it as is
+    if ! echo "$tag" | grep -qE '^[0-9]+\.[0-9]+\.[0-9]+$'; then
+        echo "$tag"
+        return
+    fi
+
+    # Try to find a related tag (release candidate or patch branch)
+    local re
+    re="${tag//./\\.}"
+    re="^$re-rc\.[0-9]+|${re%.*}.x$"
+    local resolved_tag
+    resolved_tag=$(grep -E "^$re$" "$tags" | sort -rV | head -n 1)
+    if [ -z "$resolved_tag" ] ; then
+        echo >&2 "WARNING: Could not find a matching tag for version '$ver' with tag '$tag'"
+        echo "$tag"
+        return
+    fi
+    echo "$resolved_tag"
+}
+
+# Prepare an array to store the JSON objects
+json_array=()
+
+# Read the versions and their corresponding tags from the VULNERABILITY_BUNDLE_VERSION file.
+while IFS=, read -r version tag; do
+    # Skip lines that are comments or empty
+    echo "$version" | grep -qE '^\s*(#.*|$)' && continue
+    tag=$(echo "$tag" | xargs)
+    # Check prefix
+    if [[ $tag == heads/* ]]; then
+        # Extract branch name from "heads/<branch-name>"
+        resolved_tag="${tag#heads/}"
+    elif [[ $tag == tags/* ]]; then
+        # Extract tag name from "tags/<tag-name>"
+        resolved_tag="${tag#tags/}"
+    else
+        # Resolve tag as before for tags with no prefix
+        resolved_tag=$(resolve_tag "$version" "$tag")
+    fi
```

Comment:
Assuming you changed "$tag" to "$ref".

(1.) Let's add this logic outside the `while IFS=, read -r version tag; do`:

```bash
declare -A version_map
while read -r ver tag _; do
    version_map[$ver]="$tag"
done < <(./.github/workflows/scripts/scanner-output-release-versions.sh | jq -r '.versions[] | "\(.version) \(.tag)"')
```

(2.) let's change this:

```suggestion
    # Check reference type
    case "$ref" in
        heads/*|tags/*)
            resolved_ref="${ref#*/}"
            ;;
        *)
            # Anything else we assume it is a StackRox release version
            resolved_ref=${version_map[$ref]:-}
            ;;
    esac
    if [ -z "${resolved_ref}" ]; then
        echo >&2 "error: invalid reference in VULNERABILITY_BUNDLE_VERSIONS: $ref"
        exit 1
    fi
```

And now we can close ROX-25999.